### PR TITLE
add unit tests for `fillNetConfDefaults()` in macvlan.go

### PR DESCRIPTION
**Pull request by vjayaramrh**
Signed-off-by: Vishwanath Jayaraman <vjayaram@redhat.com>

### CODE REVIEW

**Comment by vjayaramrh** [comment]
@danielmellado FYI This PR is still work in progress

File: pkg/macvlan/macvlan.go [hunk 1]
```diff
@@ -156,14 +157,14 @@ func configureIface(ifName string, res *current.Result) error {
 }
 
 func getDefaultRouteInterfaceName() (string, error) {
-	routeToDstIP, err := netlink.RouteList(nil, netlink.FAMILY_ALL)
+	routeToDstIP, err := util.GetNetLinkOps().RouteList(nil, netlink.FAMILY_ALL)
```
Comment (danielmellado) [inline pkg/macvlan/macvlan.go:160]: What's the overall idea behind moving netlink ops to an util package? Would you also be using that for mocking in the tests?

Comment (vjayaramrh) [reply pkg/macvlan/macvlan.go:160]: Yes, the approach would be similar to  https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/net_linux.go#L19-L125 and https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/net_linux_unit_test.go ... Let me know if you have different ideas

**Pull request merged**